src/settings.plugin.js

```diff
@@ -248,64 +245,73 @@ wikiaJWPlayerSettingsPlugin.prototype.updateCurrentQuality = function (data) {
 		}
 	}
 };
+// end quality
+
+// captions
+wikiaJWPlayerSettingsPlugin.prototype.onCaptionsChange = function (event) {
+	var emptyCaptionsClass = 'are-captions-empty';
+
+	clearListElement(this.captionsList);
+
+	// tracks always include "off" item
+	if (this.captionsList && event.tracks.length > 1) {
+		var suitableCaptionsTrack = this.getSuitableCaptionsIndex(
+			this.config.selectedCaptionsLanguage || this.captionLangMap[this.getUserLang()],
+			event.tracks
+		);
+
+		event.tracks.forEach(function (track, index) {
+			var captionItem = document.createElement('li'),
+				normalizedLabel = track.label === 'Off' ? 'No captions' : track.label;
+
+			captionItem.dataset.track = index;
+			captionItem.addEventListener('click', function () {
+				this.player.setCurrentCaptions(index);
+				this.adjustCaptionsPosition();
+				this.close();
+				this.player.trigger('captionsSelected', {
+					selectedLang: track.label
+				});
+			}.bind(this));
+
+			captionItem.appendChild(document.createTextNode(normalizedLabel));
+			this.captionsList.insertBefore(captionItem, this.captionsList.firstElementChild);
+		}, this);
+
+		this.wikiaSettingsElement.classList.remove(emptyCaptionsClass);
+		this.show();
 
-wikiaJWPlayerSettingsPlugin.prototype.addCaptionListener = function () {
-	var clickHandler = this.captionsClickHandler.bind(this);
-
-	this.player.on('captionsList', function (event) {
-		// tracks always include "off" item
-		if (event.tracks.length > 1) {
-			this.currentlySelectedCaptions = this.getSuitableCaptionsIndex(
-				this.captionLangMap[this.getUserLang()],
-				event.tracks
-			);
-
-			this.getLabelElement(this.captionsToggle)
-				.addEventListener('click', clickHandler);
-
-			this.wikiaSettingsElement.classList.remove('are-captions-empty');
-			this.show();
-
-			if (this.config.captions.enabled) {
-				this.player.setCurrentCaptions(this.currentlySelectedCaptions);
-			}
-
-			if (this.config.captions.styles && Object.keys(this.config.captions.styles).length) {
-				this.player.setCaptions(this.config.captions.styles);
-			}
+		// in !this.config.selectedCaptionsLanguage empty string would pass as well
```

Comment:
I don't get it.

Reply:
I removed the comment since it's not relevant now, however, I kept the `!== false` check to allow turning off the captions